router/worker.go

```diff
@@ -780,36 +782,35 @@ func (w *worker) proxyRequest(ctx context.Context, destinationJob types.Destinat
 		DestDefConfig: destination.DestinationDefinition.Config,
 		WorkspaceID:   destination.WorkspaceID,
 		DestDefName:   destination.DestinationDefinition.Name,
-		DestinationId: destination.DestinationDefinition.ID,
+		DestinationId: destination.ID,
 	}
 	ctx = context.WithValue(ctx, oauthv2.DestKey, destinationInfo)
 	oauthV2Enabled := w.rt.reloadableConfig.oauthV2Enabled.Load()
 	proxyRequestResponse := w.rt.transformer.ProxyRequest(ctx, proxyReqparams)
 	w.routerProxyStat.SendTiming(time.Since(rtlTime))
 	w.logger.Debugf(`[TransformerProxy] (Dest-%[1]v) {Job - %[2]v} Request ended`, w.rt.destType, jobID)
 	authType := oauth.GetAuthType(destinationJob.Destination.DestinationDefinition.Config)
-	var respStatusCode int
-	var respBodyTemp, contentType string
-	if proxyRequestResponse.ProxyRequestStatusCode != http.StatusOK && routerutils.IsNotEmptyString(string(authType)) && authType == oauth.OAuth && !oauthV2Enabled {
-		w.logger.Debugf(`Sending for OAuth destination`)
-		// Token from header of the request
-		respStatusCode, respBodyTemp, contentType = w.rt.handleOAuthDestResponse(&HandleDestOAuthRespParams{
-			ctx:            ctx,
-			destinationJob: destinationJob,
-			workerID:       w.id,
-			trRespStCd:     proxyRequestResponse.ProxyRequestStatusCode,
-			trRespBody:     proxyRequestResponse.ProxyRequestResponseBody,
-			secret:         m[0].Secret,
-			contentType:    proxyRequestResponse.RespContentType,
-		}, proxyRequestResponse.OAuthErrorCategory)
-
-	} else {
-		respStatusCode = proxyRequestResponse.ProxyRequestStatusCode
-		respBodyTemp = proxyRequestResponse.ProxyRequestResponseBody
-		contentType = proxyRequestResponse.RespContentType
+	if routerutils.IsNotEmptyString(string(authType)) && authType == oauth.OAuth {
```

Comment:
authType == oauth.OAuth is not sufficient?

Reply:
Sufficient